packages/web-app-files/src/components/AppBar/SelectedResources/BatchActions.vue

```diff
@@ -189,6 +244,61 @@ export default {
           })
         }
       })
+    },
+
+    acceptShares() {
+      this.triggerShareActions(shareStatus.accepted)
+    },
+
+    declineShares() {
+      this.triggerShareActions(shareStatus.declined)
+    },
+
+    triggerShareActions(newShareStatus) {
+      const errors = []
+      this.selectedFiles.forEach(async resource => {
```

Comment:
not sure if forEach can be async, even if it's resolving not parallel. P-Queue or Promise.all could be a good fit

Reply:
True, with this implementation we're silently swallowing errors. :-/ Changing to a p-queue, thanks for spotting this!